de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/wizards/new_file/NewFilePage.java

```diff
@@ -0,0 +1,114 @@
+package de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.new_file;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.eclipse.e4.core.contexts.IEclipseContext;
+
+import de.tu_bs.cs.isf.e4cf.core.gui.java_fx.util.FXMLLoader;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.controller.NewFileViewController;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.stringtable.FileTable;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.stringtable.StringTable;
+import javafx.scene.Parent;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.TextField;
+
+
+/** A wizard that lets the user create a new file. */
+public class NewFilePage {
+	private IEclipseContext context;
+	private FXMLLoader<NewFileViewController> loader;
+	public Path parentPath;
+	public Path targetPath;
+	public String inputValue;
+
+
+	// the regex to check if filename is valid.
+	private static final String FILE_NAME_REGEX = "[a-zA-Z0-9-_#]*[.]{1}[a-zA-Z0-9]+";
+	private static final String MISSIN_TYPE_REGEX = "[a-zA-Z0-9-_#]*[.]*[a-zA-Z0-9]*";
+
+	/**
+	 * @param context
+	 * @param parentPath
+	 */
+	public NewFilePage(IEclipseContext context, Path parentPath) {
+		this.context = context;
+		this.parentPath = parentPath;
+	}
+	
+	public Parent createControl(DialogPane pane) {
+		
+		final Button btnOk = (Button) pane.lookupButton(ButtonType.OK);
+		btnOk.setDisable(true);
+		loader = new FXMLLoader<NewFileViewController>(context, StringTable.BUNDLE_NAME,
+				FileTable.NEW_FILE_PAGE_VIEW);
+		
+		TextField input = loader.getController().fileNameTextField;
+		input.textProperty().addListener((obs, oldValue, newValue) -> {
+			String inputValue = input.getText();
+			if (isFilenameValid(inputValue) && isFilePathValid(inputValue)) {
+				inputValue= input.getText();
+				btnOk.setDisable(false);
+				loader.getController().alreadyExitsText.setVisible(false);

```

Comment:
Having variables for the text fields could tidy up this code

Reply:
done